Clang13 failed to compile on ubuntu20.04

Hi Sophus Team,

I'm using Clang 13.0.0 on my ubuntu 20.04 machine, with the latest commit, with the following scripts:
```
#!/bin/bash

BUILD_DIR=linux-x64
mkdir -p $BUILD_DIR
cd $BUILD_DIR
cmake ../.. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/zz/artifact/sophus/master/linux-x64 -DUSE_BASIC_LOGGING=ON
cmake --build . -j
cmake --install .
cd ..
```

My `CC` and `CXX` env vars points to my clang 13.0.0:
```
(base) zz@home% echo $CC
clang
(base) zz@home% echo $CXX
clang++
(base) zz@home% clang --version
clang version 13.0.0 (/home/zz/work/github/llvm-project dc672999a9b12a156991891dc400308b52d569ba)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/zz/soft/llvm13/bin
(base) zz@home% clang++ --version
clang version 13.0.0 (/home/zz/work/github/llvm-project dc672999a9b12a156991891dc400308b52d569ba)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/zz/soft/llvm13/bin
```

And it failed to compile with outputs like this:

<details>

```
(base) zz@home% ./linux-x64.sh 
-- The C compiler identification is Clang 13.0.0
-- The CXX compiler identification is Clang 13.0.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/zz/soft/llvm13/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/zz/soft/llvm13/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Eigen3: /usr/include/eigen3 (Required is at least version "3.3.0") 
Turning basic logging ON
-- Configuring done
-- Generating done
-- Build files have been written to: /home/zz/work/Sophus/build/linux-x64
[  4%] Building CXX object test/core/CMakeFiles/test_geometry.dir/test_geometry.cpp.o
[  8%] Building CXX object test/core/CMakeFiles/test_se3.dir/test_se3.cpp.o
[ 12%] Building CXX object test/core/CMakeFiles/test_rxso2.dir/test_rxso2.cpp.o
[ 16%] Building CXX object test/core/CMakeFiles/test_so3.dir/test_so3.cpp.o
[ 20%] Building CXX object test/core/CMakeFiles/test_so2.dir/test_so2.cpp.o
[ 25%] Building CXX object test/core/CMakeFiles/test_velocities.dir/test_velocities.cpp.o
[ 29%] Building CXX object test/core/CMakeFiles/test_sim2.dir/test_sim2.cpp.o
[ 33%] Building CXX object test/core/CMakeFiles/test_sim3.dir/test_sim3.cpp.o
[ 37%] Building CXX object test/core/CMakeFiles/test_common.dir/test_common.cpp.o
[ 41%] Building CXX object test/core/CMakeFiles/test_se2.dir/test_se2.cpp.o
[ 50%] Building CXX object examples/CMakeFiles/HelloSO3.dir/HelloSO3.cpp.o
[ 50%] Building CXX object test/core/CMakeFiles/test_rxso3.dir/test_rxso3.cpp.o
In file included from /home/zz/work/Sophus/test/core/test_so2.cpp:3:
/home/zz/work/Sophus/sophus/so2.hpp:377:15: error: definition of implicit copy assignment operator for 'SO2<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SO2(SO2 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SO2<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO2<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO2<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO2<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_so2.cpp:63:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SO2<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_so2.cpp:52:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_so2.cpp:178:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
[ 54%] Linking CXX executable HelloSO3
[ 54%] Built target HelloSO3
In file included from /home/zz/work/Sophus/test/core/test_rxso2.cpp:3:
/home/zz/work/Sophus/sophus/rxso2.hpp:440:15: error: definition of implicit copy assignment operator for 'RxSO2<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC RxSO2(RxSO2 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::RxSO2<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO2<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO2<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO2<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_rxso2.cpp:101:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::RxSO2<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_rxso2.cpp:90:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_rxso2.cpp:288:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_so3.cpp:3:
In file included from /home/zz/work/Sophus/sophus/interpolate.hpp:9:
In file included from /home/zz/work/Sophus/sophus/interpolate_details.hpp:5:
In file included from /home/zz/work/Sophus/sophus/rxso3.hpp:7:
/home/zz/work/Sophus/sophus/so3.hpp:472:15: error: definition of implicit copy assignment operator for 'SO3<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SO3(SO3 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SO3<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO3<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO3<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SO3<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_so3.cpp:78:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SO3<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_so3.cpp:66:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_so3.cpp:279:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_se2.cpp:3:
/home/zz/work/Sophus/sophus/se2.hpp:404:15: error: definition of implicit copy assignment operator for 'SE2<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SE2(SE2 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SE2<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_se2.cpp:82:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_se2.cpp:71:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_se2.cpp:249:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_se2.cpp:3:
In file included from /home/zz/work/Sophus/sophus/se2.hpp:7:
/home/zz/work/Sophus/sophus/so2.hpp:377:15: error: definition of implicit copy assignment operator for 'SO2<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SO2(SO2 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/test_se2.cpp:16:16: note: in implicit copy assignment operator for 'Sophus::SO2<double, 0>' first required here
template class SE2<double, Eigen::AutoAlign>;
               ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SE2<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_se2.cpp:82:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SE2<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_se2.cpp:71:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_se2.cpp:249:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_rxso3.cpp:3:
/home/zz/work/Sophus/sophus/rxso3.hpp:498:15: error: definition of implicit copy assignment operator for 'RxSO3<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC RxSO3(RxSO3 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::RxSO3<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO3<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO3<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::RxSO3<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_rxso3.cpp:93:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::RxSO3<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_rxso3.cpp:82:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_rxso3.cpp:305:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_se3.cpp:3:
/home/zz/work/Sophus/sophus/se3.hpp:467:15: error: definition of implicit copy assignment operator for 'SE3<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SE3(SE3 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SE3<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_se3.cpp:66:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_se3.cpp:55:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_se3.cpp:257:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
In file included from /home/zz/work/Sophus/test/core/test_se3.cpp:3:
In file included from /home/zz/work/Sophus/sophus/se3.hpp:7:
/home/zz/work/Sophus/sophus/so3.hpp:472:15: error: definition of implicit copy assignment operator for 'SO3<double, 0>' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
  SOPHUS_FUNC SO3(SO3 const& other) = default;
              ^
/home/zz/work/Sophus/test/core/test_se3.cpp:15:16: note: in implicit copy assignment operator for 'Sophus::SO3<double, 0>' first required here
template class SE3<double, Eigen::AutoAlign>;
               ^
/home/zz/work/Sophus/test/core/tests.hpp:94:18: note: in implicit copy assignment operator for 'Sophus::SE3<double, Eigen::AutoAlign>' first required here
      foo_T3_bar = foo_T_bar;
                 ^
/home/zz/work/Sophus/test/core/tests.hpp:576:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::contructorAndAssignmentTest' requested here
    passed &= contructorAndAssignmentTest();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:590:15: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doesSmallTestSetPass' requested here
    passed &= doesSmallTestSetPass();
              ^
/home/zz/work/Sophus/test/core/tests.hpp:564:12: note: in instantiation of member function 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doesLargeTestSetPass' requested here
    return doesLargeTestSetPass();
           ^
/home/zz/work/Sophus/test/core/test_se3.cpp:66:18: note: in instantiation of function template specialization 'Sophus::LieGroupTests<Sophus::SE3<double, Eigen::AutoAlign>>::doAllTestsPass<double>' requested here
    return tests.doAllTestsPass();
                 ^
/home/zz/work/Sophus/test/core/test_se3.cpp:55:19: note: in instantiation of member function 'Sophus::Tests<double>::testLieProperties' requested here
    bool passed = testLieProperties();
                  ^
/home/zz/work/Sophus/test/core/test_se3.cpp:257:19: note: in instantiation of member function 'Sophus::Tests<double>::runAll' requested here
  Tests<double>().runAll();
                  ^
^Cmake[2]: *** [test/core/CMakeFiles/test_se2.dir/build.make:76: test/core/CMakeFiles/test_se2.dir/test_se2.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:430: test/core/CMakeFiles/test_se2.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_velocities.dir/build.make:76: test/core/CMakeFiles/test_velocities.dir/test_velocities.cpp.o] 中断
make[2]: *** [test/core/CMakeFiles/test_geometry.dir/build.make:76: test/core/CMakeFiles/test_geometry.dir/test_geometry.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:196: test/core/CMakeFiles/test_geometry.dir/all] 中断
make[1]: *** [CMakeFiles/Makefile2:274: test/core/CMakeFiles/test_velocities.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_so3.dir/build.make:76: test/core/CMakeFiles/test_so3.dir/test_so3.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:352: test/core/CMakeFiles/test_so3.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_sim3.dir/build.make:76: test/core/CMakeFiles/test_sim3.dir/test_sim3.cpp.o] 中断
make[2]: *** [test/core/CMakeFiles/test_se3.dir/build.make:76: test/core/CMakeFiles/test_se3.dir/test_se3.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:378: test/core/CMakeFiles/test_sim3.dir/all] 中断
make[1]: *** [CMakeFiles/Makefile2:248: test/core/CMakeFiles/test_se3.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_rxso2.dir/build.make:76: test/core/CMakeFiles/test_rxso2.dir/test_rxso2.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:222: test/core/CMakeFiles/test_rxso2.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_sim2.dir/build.make:76: test/core/CMakeFiles/test_sim2.dir/test_sim2.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:326: test/core/CMakeFiles/test_sim2.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_common.dir/build.make:76: test/core/CMakeFiles/test_common.dir/test_common.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:404: test/core/CMakeFiles/test_common.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_rxso3.dir/build.make:76: test/core/CMakeFiles/test_rxso3.dir/test_rxso3.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:456: test/core/CMakeFiles/test_rxso3.dir/all] 中断
make[2]: *** [test/core/CMakeFiles/test_so2.dir/build.make:76: test/core/CMakeFiles/test_so2.dir/test_so2.cpp.o] 中断
make[1]: *** [CMakeFiles/Makefile2:300: test/core/CMakeFiles/test_so2.dir/all] 中断
make: *** [Makefile:146: all] 中断

```

</details>

--- a/sophus/rxso2.hpp
+++ b/sophus/rxso2.hpp
@@ -425,6 +425,11 @@ class RxSO2 : public RxSO2Base<RxSO2<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC RxSO2& operator=(RxSO2 const& other) = default;
+
   static int constexpr DoF = Base::DoF;
   static int constexpr num_parameters = Base::num_parameters;
 

--- a/sophus/rxso3.hpp
+++ b/sophus/rxso3.hpp
@@ -485,6 +485,11 @@ class RxSO3 : public RxSO3Base<RxSO3<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC RxSO3& operator=(RxSO3 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes quaternion to identity rotation and scale

--- a/sophus/se2.hpp
+++ b/sophus/se2.hpp
@@ -393,6 +393,11 @@ class SE2 : public SE2Base<SE2<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC SE2& operator=(SE2 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes rigid body motion to the identity.

--- a/sophus/se3.hpp
+++ b/sophus/se3.hpp
@@ -456,6 +456,11 @@ class SE3 : public SE3Base<SE3<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC SE3& operator=(SE3 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes rigid body motion to the identity.

--- a/sophus/sim2.hpp
+++ b/sophus/sim2.hpp
@@ -393,6 +393,11 @@ class Sim2 : public Sim2Base<Sim2<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC Sim2& operator=(Sim2 const& other) = default;
+
   static int constexpr DoF = Base::DoF;
   static int constexpr num_parameters = Base::num_parameters;
 

--- a/sophus/sim3.hpp
+++ b/sophus/sim3.hpp
@@ -394,6 +394,11 @@ class Sim3 : public Sim3Base<Sim3<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC Sim3& operator=(Sim3 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes similarity transform to the identity.

--- a/sophus/so2.hpp
+++ b/sophus/so2.hpp
@@ -366,6 +366,11 @@ class SO2 : public SO2Base<SO2<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC SO2& operator=(SO2 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes unit complex number to identity rotation.

--- a/sophus/so3.hpp
+++ b/sophus/so3.hpp
@@ -460,6 +460,11 @@ class SO3 : public SO3Base<SO3<Scalar_, Options>> {
 
   using Base::operator=;
 
+  /// Define copy-assignment operator explicitly. The definition of
+  /// implicit copy assignment operator is deprecated in presence of a
+  /// user-declared copy constructor (-Wdeprecated-copy in clang >= 13).
+  SOPHUS_FUNC SO3& operator=(SO3 const& other) = default;
+
   EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
   /// Default constructor initializes unit quaternion to identity rotation.

